copylot/hardware/cameras/abstract_camera.py

```diff
@@ -1,5 +1,113 @@
-import abc
+from abc import ABCMeta, abstractmethod
 
 
-class AbstractCamera(metaclass=abc.ABCMeta):
-    pass
+class AbstractCamera(metaclass=ABCMeta):
+
+    @abstractmethod
+    def snap(self):
+        """Method to capture a single image from the camera."""
+        raise NotImplementedError()
+
+    @abstractmethod
+    def list_available_cameras(self):
+        """List all cameras that driver discovers."""
+        raise NotImplementedError()
```

Comment:
It feels surprising that this method would be an instance method of a particular camera (in other words, why is listing all cameras a function of `self`)?

Reply:
yes this could be static